app/src/main/java/com/example/hris/ui/fragments/timelogs/TimeLogsFragment.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.hris.ui.fragments.timelogs
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.example.hris.R
+import com.example.hris.databinding.FragmentTimeLogsBinding
+import com.example.hris.ui.adapters.TimeLogsListAdapter
+import com.example.hris.ui.viewmodels.MainActivityViewModel
+import com.example.hris.ui.viewmodels.TimeLogsViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class TimeLogsFragment : Fragment() {
+    private val viewModel: TimeLogsViewModel by viewModels()
+    private val mainViewModel: MainActivityViewModel by activityViewModels()
+    private lateinit var binding: FragmentTimeLogsBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentTimeLogsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.timelogsToolbar.btnAdd.setOnClickListener {
+            findNavController().navigate(R.id.action_timeLogsFragment_to_addTimeLogsFragment)
+        }
+
+        viewModel.user.observe(viewLifecycleOwner) {
+            viewModel.callTimeLogs()
+        }
```

Comment:
Why do you need to observe the `user` live data before calling `callTimeLogs()`? Are there instances where the contents of the `User` object change while inside the `TimeLogsFragment`?

Reply:
No sir, pero there are instances where using the `userData` value in the repository throws a nullpointer exception although the value of `userData` has been added in the previous activity